src/components/NanoContract/NanoContractTransactionsList.component.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { useNavigation } from '@react-navigation/native';
+import React, { useState } from 'react';
+import {
+  StyleSheet,
+  View,
+} from 'react-native';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { NanoContractTransactionsListHeader } from './NanoContractTransactionsListHeader.component';
+import { NanoContractTransactionsListItem } from './NanoContractTransactionsListItem.component';
+import { COLORS } from '../../styles/themes';
+import { nanoContractAddressChangeRequest } from '../../actions';
+import { HathorFlatList } from '../HathorFlatList';
+
+const getNanoContractHistory = (ncId) => (state) => {
+  const history = state.nanoContract.history[ncId];
+  return Object.values(history || {});
+}
+
+/**
+ * It presents a list of transactions from selected Nano Contract.
+ *
+ * @param {Object} props
+ * @param {Object} props.nc Nano Contract data
+ * @param {string} props.nc.ncId Nano Contract ID
+ * @param {string} props.nc.address Default caller address for Nano Contract interaction
+ */
+export const NanoContractTransactionsList = ({ nc }) => {
```

Comment:
I think this name is misleading. I propose `NanoContractDetails` because it's not only a list of transactions.

Reply:
Humm... Ok. But it will drive a change in the name of the screen as well. I do it.

Reply:
Also done.